services/checks/checks.go

```diff
@@ -316,26 +319,82 @@ func (s *Service) executeMongoDBChecks(ctx context.Context, checks []check.Check
 				continue
 			}
 
-			nCtx, cancel := context.WithTimeout(ctx, resultTimeout)
-			_, err := s.waitForResult(nCtx, r.ID) // TODO returns result
-			cancel()
-			if err != nil {
-				s.l.Errorf("failed to get check result: %s", err)
-				continue
+			if err = s.processResults(ctx, c, target, r.ID); err != nil {
+				s.l.Errorf("failed to process action result: %s", err)
 			}
+		}
+	}
+
+	return nil
+}
+
+// TODO find better name
+func (s *Service) processResults(ctx context.Context, check check.Check, target target, resID string) error {
+	nCtx, cancel := context.WithTimeout(ctx, resultTimeout)
+	r, err := s.waitForResult(nCtx, resID)
+	cancel()
+	if err != nil {
+		return errors.Wrap(err, "failed to get check result")
+	}
 
-			// TODO process result
+	funcs, err := getFuncsForVersion(check.Version)
+	if err != nil {
+		return err
+	}
+
+	env, err := starlark.NewEnv(check.Name, check.Script, funcs)
+	if err != nil {
+		return errors.Wrap(err, "failed to prepare starlark environment")
+	}
+
+	results, err := env.Run(check.Name, r, s.l.Debug)
+	if err != nil {
+		return errors.Wrap(err, "failed to execute script")
+	}
+
+	for _, result := range results {
+		alert, err := makeAlert(check.Name, target, &result)
+		if err != nil {
+			return errors.Wrap(err, "failed to create alert")
 		}
+
+		s.alertsRegistry.Add(check.Name, time.Second, alert)
 	}
 
 	return nil
 }
 
+func makeAlert(name string, target target, result *check.Result) (*ammodels.PostableAlert, error) {
+	labels, err := models.MergeLabels(target.node, target.service, target.agent)
+	if err != nil {
+		return nil, err
+	}
+
+	labels[model.AlertNameLabel] = name
+	labels["severity"] = result.Severity.String()
+	labels["stt_check"] = "1"
+
+	return &ammodels.PostableAlert{
+		Alert: ammodels.Alert{
+			// GeneratorURL: "TODO",
+			Labels: labels,
+		},
+
+		// StartsAt and EndAt can't be added there without changes in Registry
+
+		Annotations: map[string]string{
+			"summary":     result.Summary,
+			"description": result.Description,
+		},
+	}, nil
+}
+
 // target contains required info about check target
 type target struct {
-	agentID   string
-	serviceID string
-	dsn       string
+	agent   *models.Agent
+	service *models.Service
+	node    *models.Node
```

Comment:
I think both `agent` and `node` are not actually used – they were added for demo.

Reply:
All that stuff needed to create labels set. Refactored, now targen contains labels set instead of node/service/agent.